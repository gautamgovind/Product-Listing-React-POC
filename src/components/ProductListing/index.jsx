import React from 'react';
import { useSelector } from "react-redux";

import Card from '../Card';
import { ProductListContainer } from './index.style';

const ProductListing = () => {
    const products = useSelector(state=>state.productListReducer.products);

    return (
        <ProductListContainer>
            {products.map(product => <Card key={`product-${product.index}`} product={product} /> )}
        </ProductListContainer>
    );
}

export default ProductListing;