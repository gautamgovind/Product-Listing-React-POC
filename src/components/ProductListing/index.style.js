import styled from "styled-components";

export const ProductListContainer = styled.ul`
    display: flex;
    flex-wrap: wrap;
    list-style: none;
    margin: 0;
    padding: 0;
`;