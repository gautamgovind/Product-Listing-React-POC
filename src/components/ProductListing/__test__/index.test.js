import React from 'react';
import { shallow } from "enzyme";

import ProductListing from "../index";

jest.mock('react-redux', () => ({
    useDispatch: () => {},
    useSelector: () => ([ {
        index: 0,
        isSale: true,
        isExclusive: true,
        price: "$18.88",
        productName: "Striped shirt",
        size: [ "XS", "S", "L", "XL" ]
    }]),
  })); 

describe('<ProductListing />', function () {
 
  it('renders without crashing', function () {
    const wrapper = shallow(<ProductListing />);
    expect(wrapper).toMatchSnapshot();
  });
});