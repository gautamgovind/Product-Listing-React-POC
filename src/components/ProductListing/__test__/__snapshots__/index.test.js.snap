// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductListing /> renders without crashing 1`] = `ShallowWrapper {}`;
